distributed/shuffle/tests/test_merge.py

```diff
@@ -106,6 +107,49 @@ async def test_basic_merge(c, s, a, b, how, lose_annotations):
     )
 
 
+@gen_cluster(client=True)
+async def test_merge_p2p_shuffle_reused_dataframe_with_different_parameters(c, s, a, b):
```

Comment:
This test passes for me on main. From what I can tell, this is expected, isn't it?

Reply:
Yes, there's no reuse on `main`. This test makes sure that we don't reuse too much (which is what originally happened in `dask-expr`).